### PR TITLE
Add service tagging details to network costs

### PR DESCRIPTION
**Pull request by srpomeroy**

### CODE REVIEW

**Comment by srpomeroy** [comment]
@mbolt35 Can you give a quick review of the accuracy of the service level tagging?

File: architecture.md [hunk 1]
```diff
@@ -8,14 +8,14 @@ Below are the major components deployed with the [Kubecost helm chart](http://do
     b. Cost-model -- provides cost allocation calculations and metrics, both reads and writes to Prometheus  
 2. **Prometheus**  
     a. Prometheus server -- time-series data store for cost & health metrics  
-    b. Kube-state-metrics -- provides Kubernetes API metrics, e.g. resource requests  
-    c. Node-exporter -- provides node-level utilization metrics for right-sizing recommendations and cluster utilization
```
Comment (dwbrown2) [inline architecture.md:12]: @srpomeroy I believe NE still used for autoscaling cluster right-sizing? Otherwise this LGTM!

Comment (srpomeroy) [reply architecture.md:12]: @dwbrown2 There is much confusion around NE and where it is actually needed. We recently added a resource tuning guide to our docs. [There is a section on NE](https://guide.kubecost.com/hc/en-us/articles/6446286863383-Tuning-Resource-Consumption#disable-or-stop-scraping-node-exporter) that claims disabling it only impacts cluster health. Is that not accurate?

Comment (mbolt35) [reply architecture.md:12]: @srpomeroy That doesn't appear accurate based on just an issue I looked into the other day: https://github.com/kubecost/cost-analyzer-frontend/issues/1023#issuecomment-1231989077

These look like node-exporter metrics yeah?
```
query=sum(node_memory_MemTotal_bytes - node_memory_MemFree_bytes - node_memory_Cached_bytes - node_memory_Buffers_bytes) / 1024 / 1024 / 1024
```

I have brought this up multiple times that there are likely a handful of node exporter queries that could effect more parts of the application. They are almost all legacy features and are somewhat difficult to track down until they pop up in bug tickets.

**Comment by mbolt35** [comment]
> @mbolt35 Can you give a quick review of the accuracy of the service level tagging?

@srpomeroy What you have sounds accurate. If you want to add any further details, we could probably add the source of this information (since it doesn't require cloud integration with kubecost to use):

* AWS IP Ranges: https://ip-ranges.amazonaws.com/ip-ranges.json
* GCP, we use their service tags:
```
// GoogleServiceTags contains a listing of all gcloud subdomains used in the gcloud go api.
var GoogleServiceTags []string = []string{
	"accessapproval",
	"accesscontextmanager",
	"aiplatform",
	"analyticsadmin",
	"analyticsdata",
	"apigateway",
	"apigeeconnect",
	"appengine",
	"area120tables",
	"artifactregistry",
	"cloudasset",
	"assuredworkloads",
	"automl",
	"bigqueryconnection",
	"bigquerydatatransfer",
	"bigquerymigration",
	"bigqueryreservation",
	"bigquerystorage",
	"cloudbilling",
	"billingbudgets",
	"binaryauthorization",
	"cloudchannel",
	"cloudbuild",
	"datamigration",
	"cloudtasks",
	"compute",
	"contactcenterinsights",
	"container",
	"containeranalysis",
	"datacatalog",
	"dataflow",
	"datafusion",
	"datalabeling",
	"dataproc",
	"dataqna",
	"datastore",
	"datastream",
	"clouddebugger",
	"clouddeploy",
	"dialogflow",
	"dlp",
	"documentai",
	"domains",
	"clouderrorreporting",
	"essentialcontacts",
	"eventarc",
	"file",
	"firestore",
	"cloudfunctions",
	"gameservices",
	"connectgateway",
	"gkehub",
	"gsuiteaddons",
	"iamcredentials",
	"iap",
	"ids",
	"cloudiot",
	"cloudkms",
	"language",
	"lifesciences",
	"logging",
	"longrunning",
	"managedidentities",
	"mediatranslation",
	"memcache",
	"metastore",
	"monitoring",
	"networkconnectivity",
	"networkmanagement",
	"networksecurity",
	"notebooks",
	"composer",
	"orgpolicy",
	"osconfig",
	"oslogin",
	"phishingprotection",
	"policytroubleshooter",
	"cloudprivatecatalog",
	"pubsub",
	"pubsublite",
	"recaptchaenterprise",
	"recommendationengine",
	"recommender",
	"redis",
	"cloudresourcemanager",
	"resourcesettings",
	"retail",
	"cloudscheduler",
	"secretmanager",
	"privateca",
	"securitycenter",
	"servicecontrol",
	"servicedirectory",
	"servicemanagement",
	"serviceusage",
	"cloudshell",
	"spanner",
	"speech",
	"storage",
}
```

We convert each of the tags + `".googleapis.com"` into an IP Range.
I think we also cross-reference some of the GCP IP ranges with the following URLs as well:
* https://www.gstatic.com/ipranges/goog.json
* https://www.gstatic.com/ipranges/cloud.json

For Azure, I just noticed that they rotate their IP Range files every so often and reset the old download link. We _were_ using:
* https://download.microsoft.com/download/7/1/D/71D86715-5596-4529-9B13-DA13A5DE5B63/ServiceTags_Public_20211206.json

But this now 404s. The new link is:
* https://download.microsoft.com/download/7/1/D/71D86715-5596-4529-9B13-DA13A5DE5B63/ServiceTags_Public_20220829.json

But this isn't in the current version of network-costs. Will probably need to create a ticket for this. I need to figure out how to get this data after it's updated.

File: architecture.md (hunk 1 above)
Comment (dwbrown2) [reply architecture.md:12]: Here are all the areas that I'm aware of:

* Reserved Instance Recommendations in Savings
* Various Kubecost Grafana dashboards
* Several health alerts

None of these are critical to core Kubecost. Are we missing any?

Before redoing overview, we closely tested without NE to confirm and came up with these: https://guide.kubecost.com/hc/en-us/articles/4407595942679-Kubecost-Diagnostics#node-exporter-metrics-available

They should be updated. Let me know if I can help with this. Also, cc @bstuder99 for docs!

**Pull request merged**